lib/tasks/dev.rake

```diff
@@ -0,0 +1,51 @@
+# Don't add any includes that aren't in the Ruby std-lib here,
```

Comment:
I understand, and **I'm not sure I see the duplication that you do.**

The rake task intends to handle only Rails-oriented environment setup. It does five main things:

1. Copy required Rails config files into place
2. Install required Ruby gems using the `bundle` command
3. Set up the required databases for the Rails app (create, migrate, and seed)
4. Generate Sphinx configuration for the Rails app (first checking that Sphinx is installed)
5. Generate Sphinx indexes for the Rails app

When I look at `bootstrap.sh`, the only overlap I see between this rake task and `bootstrap.sh` is copying the proper database configuration file in place. **Perhaps I'm not seeing other parts which overlap. Could you say more about this?**

Meanwhile, I think we could separate concerns into two here:

1. **Setting up the development machine (installing packages like mysql and sphinx, configuring the system, etc.)** This should definitely be handled by `bootstrap.sh` for the OpenSUSE vagrant-based development environment.
2. **Setting up the Rails app and Rails environment.** This could be handled _for all environments_ by the rake task here.

Doing this could DRY up a couple of things:

* `bootstrap.sh` could call the rake task once the development machine setup is complete
* We could replace the entire `before_script` section in `travis.yml` with `bundle exec rake dev:bootstrap`

I'm happy to extend this PR to cover those changes, as well as updating the README, if you agree with this direction.